packetbeat/protos/http/http_parser.go

```diff
@@ -375,28 +379,39 @@ func (parser *parser) parseHeader(m *message, data []byte) (bool, bool, int) {
 }
 
 func (*parser) parseBody(s *stream, m *message) (ok, complete bool) {
-	if isDebug {
-		debugf("parseBody body: %d", s.parseOffset)
-	}
+	nbytes := len(s.data)
 	if !m.hasContentLength && (bytes.Equal(m.connection, constClose) ||
 		(isVersion(m.version, 1, 0) && !bytes.Equal(m.connection, constKeepAlive))) {
 
+		m.size += uint64(nbytes)
+		s.bodyReceived += nbytes
+		m.contentLength += nbytes
+
 		// HTTP/1.0 no content length. Add until the end of the connection
 		if isDebug {
-			debugf("http conn close, received %d", len(s.data)-s.parseOffset)
+			debugf("http conn close, received %d", len(s.data))
 		}
-		s.bodyReceived += (len(s.data) - s.parseOffset)
-		m.contentLength += (len(s.data) - s.parseOffset)
-		s.parseOffset = len(s.data)
+		if m.saveBody {
+			m.body = append(m.body, s.data...)
```

Comment:
Is there a chance `len(body) > http header content length` after this append? E.g. another HTTP message starting right after the current one due to HTTP keep-alive/pipelining.

Reply:
Not in this particular append, as it is only called in the particular case where the length is not known and connection is `close` or http/1.0 is used, which rules out keep-alive/pipelining.